apps/ucp-ui/src/ApiKeys/ApiKeys.tsx

```diff
@@ -65,34 +86,29 @@ const ApiKeys = () => {
           }}
         />
         <Divider />
-        <CardContent>
-          {!hasApiKeyAccess && (
-            <Stack spacing={0.5}>
-              <Typography variant="h5">
-                {API_KEYS_REQUEST_ACCESS_TITLE_TEXT}
-              </Typography>
-              <Typography className={styles.secondaryColor} variant="body1">
-                To access UCP services, you will need API keys. To request your
-                keys, please email us at {SUPPORT_EMAIL}. We will process your
-                request promptly. Once your access is approved, you can return
-                here to generate your keys. If you have already submitted a
-                request, please check back later.
-              </Typography>
-            </Stack>
-          )}
-        </CardContent>
-        <CardActions className={styles.cardActions}>
-          {!hasApiKeyAccess ? (
-            <Button
-              href={`mailto:${SUPPORT_EMAIL}?subject=API Keys Request for ${user?.email}&body=${newLine}${newLine}----------Do not edit anything below this line----------${newLine}User Email: ${user?.email}`}
-              size="large"
-              startIcon={<MailOutline />}
-              variant="contained"
-            >
-              {REQUEST_API_KEY_ACCESS_BUTTON_TEXT}
-            </Button>
-          ) : null}
-        </CardActions>
+        {!hasApiKeyAccess && <RequestAPIKeyAccess />}
+        {canUserGenerateApiKeys && <GenerateKeys />}
+        {shouldShowGetApiKeysError && (
+          <CardContent>
+            <FetchError
+              description={API_KEYS_GET_KEYS_FAILURE_TEXT}
+              refetch={() => void refetchApiKeys()}
+              title="Something went wrong"
+            />
+          </CardContent>
+        )}
+        {hasKeys && (
+          <CardContent>
+            <TextField
+              InputProps={{
+                readOnly: true,
+              }}
+              label={API_KEYS_CLIENT_ID_LABEL_TEXT}
+              type="password"
+              value={clientId}
+            />
```

Comment:
I only put this here so that I could write an e2e test without having to rewrite how the e2e test was done when we had keys to show. I'm intentionally leaving the secret out until the next ticket